crates/red_knot_python_semantic/src/types.rs

```diff
@@ -74,72 +74,75 @@ pub fn check_types(db: &dyn Db, file: File) -> TypeCheckDiagnostics {
 }
 
 /// Infer the public type of a symbol (its type as seen from outside its scope).
-#[salsa::tracked]
-fn symbol_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymbolId) -> Symbol<'db> {
-    let _span = tracing::trace_span!("symbol_by_id", ?symbol).entered();
-
-    let use_def = use_def_map(db, scope);
-
-    // If the symbol is declared, the public type is based on declarations; otherwise, it's based
-    // on inference from bindings.
-
-    let declarations = use_def.public_declarations(symbol);
-    let declared = declarations_ty(db, declarations);
-
-    match declared {
-        // Symbol is declared, trust the declared type
-        Ok(symbol @ Symbol::Type(_, Boundness::Bound)) => symbol,
-        // Symbol is possibly declared
-        Ok(Symbol::Type(declared_ty, Boundness::PossiblyUnbound)) => {
-            let bindings = use_def.public_bindings(symbol);
-            let inferred = bindings_ty(db, bindings);
-
-            match inferred {
-                // Symbol is possibly undeclared and definitely unbound
-                Symbol::Unbound => {
-                    // TODO: We probably don't want to report `Bound` here. This requires a bit of
-                    // design work though as we might want a different behavior for stubs and for
-                    // normal modules.
-                    Symbol::Type(declared_ty, Boundness::Bound)
+fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
+    #[salsa::tracked]
+    fn symbol_by_id<'db>(
+        db: &'db dyn Db,
+        scope: ScopeId<'db>,
```

Comment:
I tried to remove this (and use `scope` from the outer scope), but it seems to be required in order to make this salsa_tracked.

Reply:
This works for me. Thanks